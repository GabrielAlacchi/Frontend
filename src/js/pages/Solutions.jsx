import React from "react"
import Textarea from "react-textarea-autosize";

export default class Solutions extends React.Component {

    problem = [];
    problem = this.props.location.state.testvalue;

    constructor(props) {
        super(props);
        this.state = {value: ''};

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        this.setState({value: event.target.value});
    }

    handleSubmit(event) {
        if (this.state.value == ""){
            alert("Please enter a value");

        }else {
            alert(this.state.value);
            //backend code
        }
        event.preventDefault();
    }

    render() {

        return(
            <div>
              <h1>{this.problem.name}</h1>
                <p>{this.problem.description}</p>
                <p>Enter solution in the box below</p>
                <Textarea style = {{width:900, height: 300}} onChange={this.handleChange}/>
                <a class="btn -btn-default" onClick={this.handleSubmit}>Submit</a>
            </div>
        );
    }
}