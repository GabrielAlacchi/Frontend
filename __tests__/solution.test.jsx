import React from 'react';
import Solutions from '../src/js/pages/Solutions.jsx';
import renderer from 'react-test-renderer';

test('Solution page renders correct information',() =>{
    const component = renderer.create(
        <Solutions/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})