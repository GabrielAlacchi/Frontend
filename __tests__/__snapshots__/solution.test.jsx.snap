// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Solution page renders correct information 1`] = `
<div>
    <h1>
        {this.problem.name}
    </h1>
    <p>
        {this.problem.description}
    </p>
    <p>
        Enter solution in the box below
    </p>
    <
        Textarea style = {{width:900, height: 300}}
        onChange={[Function]}
    />
    <a
        className="btn -btn-default"
        onClick={[Function}}
    >
        Submit
    </a>
    <button
        onClick={[Function]}
    >
        Success Demo
    </button>
    <button
        onClick={[Function}]}
    >Fail Demo
    </button>
    <button
        onClick={[Function]}
    >
        Save Progress
    </button>
</div>
`;